core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -725,9 +722,15 @@ object SparkSubmit extends CommandLineUtils {
       printWarning("Subclasses of scala.App may not work correctly. Use a main() method instead.")
     }
 
-    val mainMethod = mainClass.getMethod("main", new Array[String](0).getClass)
-    if (!Modifier.isStatic(mainMethod.getModifiers)) {
-      throw new IllegalStateException("The main method in the given main class must be static")
+    val sparkAppMainMethod = mainClass.getMethods().find(_.getName == "sparkMain")
+    val childSparkConf = sysProps.filter{ p => p._1.startsWith("spark.") }.toMap
```

Comment:
nit: again, please address the feedback that is given. I've lost count of how many times I've pointed out that there's a missing space between `filter` and `{`.